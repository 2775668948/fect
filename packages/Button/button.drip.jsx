import {
  defineComponent,
  computed,
  ref,
  toRefs,
  onMounted,
  watchEffect,
  watch,
} from 'vue'

//animation default style
const DEFAULT_STYLE = {
  left: '',
  top: '',
  opacity: 0.5,
  transform: 'translate(-50%, -50%)',
}

const ButtonDrip = defineComponent({
  props: {
    onCompleted: Function,
  },
  setup(props, { attrs, slots, emit }) {
    const show = ref(false)
    const dripRef = ref(null)
    const { onCompleted } = toRefs(props)
    // onCompleted.value()
    const styles = computed(() => {
      const styles = DEFAULT_STYLE
      return styles
    })
    return () => (
      <>
        <span
          ref={dripRef}
          className={'fay-button-drip'}
          style={styles.value}
          v-show={show.value}
        ></span>
        {/* <style jsx>{`
          .fay-button-drip {
          }
        `}</style> */}
      </>
    )
  },
})

export default ButtonDrip
