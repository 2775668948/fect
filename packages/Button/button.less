@colors : red;

.fay-btn{
  box-sizing: border-box;
  display: inline-block;
  // padding: ;
  font-weight: 400;
  user-select: none;
  text-transform: capitalize;
  white-space: nowrap;
  text-align: center;
  outline: none;
  width: auto;
  border: 1px solid var(--body-colors);
}