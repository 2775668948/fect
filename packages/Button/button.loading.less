.fect-btn-loading {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 2;
  &.success {
    border-color: var(--geist-success);
    background-color: var(--geist-success);
    color: #fff;
  }
  &.warning {
    border-color: var(--geist-warning);
    background-color: var(--geist-warning);
    color: #fff;
  }
  &.error {
    background-color: var(--geist-error);
    border-color: #fff;
    color: #fff;
  }
}
