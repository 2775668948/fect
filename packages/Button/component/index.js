import Button from '.'

// Button.install = vue =>vue.component(Button.name,Button)
export default Button
