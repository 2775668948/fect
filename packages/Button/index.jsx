import { defineComponent, toRefs } from 'vue'
import validator from '../utils/validator'
import theme from '../utils/theme'
import './button.less'
const { buttonTypes,normalSizes } = theme

const Buttton = defineComponent({
  props:{
    type:{
      type:String,
      validator:validator.enums(buttonTypes),
    },
    size:{
      type:String,
      validator:validator.enums(normalSizes),
    },
    disabled:Boolean,
    shadow:Boolean,
    loading:Boolean,
  },
  emits:['click'],
  setup(props,{ attrs,slots,emit }){
    const { size,type } = toRefs(props)

    const clickHandler = (e)=>emit('click',e)
    return ()=>
      ( <>
        <div 
          className={'fay-btn'}
          onClick={clickHandler}>{slots && slots.default()}</div>
      </>)
  },
})


export default Buttton
