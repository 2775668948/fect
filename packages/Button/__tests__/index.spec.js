import { mount } from '@vue/test-utils'
import Button from '../index'
import { ref } from 'vue'

describe('Button', () => {
  it('should be render as Element', () => {
    const button = mount(Button)
    expect(button.html()).toMatchSnapshot()
  })

  it('should emit click event', () => {
    const button = mount(Button)
    button.trigger('click')
    expect(button.emitted('click').length).toEqual(1)
  })

  it('should no emit click event when status as disabled', () => {
    const wrapper = mount(Button, {
      props: {
        disabled: true,
      },
    })
    expect(wrapper.emitted()).toEqual({})
  })

  it('should no emit click event when sstatus as loading', () => {
    const wrapper = mount(Button, {
      props: {
        loading: true,
      },
    })
    expect(wrapper.emitted()).toEqual({})
  })

  it('should allowed all types', () => {
    const wrapper = mount({
      setup(props) {
        const typeList = ['default', 'success', 'warning', 'error']
        return {
          typeList,
        }
      },
      render(h) {
        return (
          <>
            {this.typeList.map((item) => (
              <Button type={item} />
            ))}
          </>
        )
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should allowed all size', () => {
    const wrapper = mount({
      setup(props) {
        const sizeList = ['mini', 'small', 'medium', 'large']
        return {
          sizeList,
        }
      },
      render(h) {
        return (
          <>
            {this.sizeList.map((item) => (
              <Button type={item} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })

  it('should be support default render', async () => {
    const wrapper = mount({
      setup() {
        const val = ref(0)
        const handlerClick = () => (val.value = 1)
        return { val, handlerClick }
      },
      render() {
        return <Button onClick={this.handlerClick}>{this.val}</Button>
      },
    })

    expect(wrapper.find('.fay-btn').text()).toBe('0')
    await wrapper.find('.fay-btn').trigger('click')
    expect(wrapper.find('.fay-btn').text()).toBe('1')
  })
})
