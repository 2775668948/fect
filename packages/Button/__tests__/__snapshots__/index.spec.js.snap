// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should allowed all size 1`] = `
<button class="fay-btn mini medium ">
  <!---->
  <!---->
  <!---->
</button><button class="fay-btn small medium ">
  <!---->
  <!---->
  <!---->
</button><button class="fay-btn medium medium ">
  <!---->
  <!---->
  <!---->
</button><button class="fay-btn large medium ">
  <!---->
  <!---->
  <!---->
</button>
`;

exports[`Button should allowed all types 1`] = `
<button class="fay-btn default medium ">
  <!---->
  <!---->
  <!---->
</button><button class="fay-btn success medium ">
  <!---->
  <!---->
  <!---->
</button><button class="fay-btn warning medium ">
  <!---->
  <!---->
  <!---->
</button><button class="fay-btn error medium ">
  <!---->
  <!---->
  <!---->
</button>
`;

exports[`Button should be render as Element 1`] = `
<button class="fay-btn default medium ">
  <!---->
  <!---->
  <!---->
</button>
`;
